mslib/utils/view_restoration.py

```diff
@@ -0,0 +1,162 @@
+                logging.warning("Corrupted view_settings.json, initializing new file")
+
+        settings_data[flight_track_name] = {
+            "global": global_data,
+            "views": settings
+        }
+        with settings_file.open("w", encoding="utf-8") as f:
+            json.dump(settings_data, f, indent=2)
+        return True
+    except Exception as e:
+        logging.error("Failed to save view settings for %s: %s", flight_track_name, str(e))
+        return False
+
+
+def set_global_data(flight_track=None):
+    """
+    Create the global section for view settings using waypoints from flight_track or topview.
+    """
+    if flight_track:
+        flight_track_name = flight_track.name
+    else:
+        logging.warning("No flight track provided; using default flight track name")
+
+    return {
+        "mss_version": str(mss_version),
+        "flight_track_name": str(flight_track_name)
+    }
+
+
+def serialize_settings(settings_list):
+    """
+    Recursively serialize a list of settings dictionaries to make them JSON-serializable.
+    """
+    if isinstance(settings_list, dict):
+        settings_list = [settings_list]  # Handle single dict case
+    serialized_list = []
+    for settings in settings_list:
+        if not isinstance(settings, dict):
+            logging.warning("Invalid settings entry skipped: %s", settings)
+            continue
+        serialized = {}
+        for key, value in settings.items():
+            if hasattr(value, '__dict__'):
+                serialized[key] = {
+                    attr: getattr(value, attr)
+                    for attr in dir(value)
+                    if not attr.startswith('_') and isinstance(
+                        getattr(value, attr), (str, int, float, bool, list, dict, type(None))
+                    )
+                }
+            elif isinstance(value, QtCore.QDateTime):
+                serialized[key] = value.toString(QtCore.Qt.ISODate)
+            elif isinstance(value, dict):
+                serialized[key] = serialize_settings([value])[0]
+            elif isinstance(value, list):
+                serialized[key] = serialize_settings(value)
+            elif isinstance(value, (tuple, QtGui.QColor)):
+                serialized[key] = list(value)
+            else:
+                serialized[key] = value
+        serialized_list.append(serialized)
+    return serialized_list if isinstance(settings_list, list) else serialized_list[0]
+
+
+def restore_view_settings(flight_track_name):
+    """
+    Restore view settings from the JSON file and adjust waypoints' flightlevel.
+    """
+    default_settings = {
+        "global": {"mss_version": str(mss_version), "flight_track_name": "Unknown"},
+        "views": []
+    }
+
+    if not isinstance(flight_track_name, str):
+        logging.warning("Invalid flight_track_name '%s'; converting to string", flight_track_name)
+        flight_track_name = str(flight_track_name)
```

Comment:
In what situation does this actually happen, and shouldn't it be the responsibility of the caller to provide the correct argument type?